src/components/movie-card/movie-card.jsx

```diff
@@ -1,14 +1,85 @@
 import PropTypes from "prop-types";
 import { Button, Card } from "react-bootstrap";
+import { Link } from "react-router-dom";
+import { useState } from "react";
+
+export const MovieCard = ({ movie, onToggleFavorite }) => {
+  const token = localStorage.getItem("token");
+  const [isFavorite, setIsFavorite] = useState(false);
+  const storedUser = JSON.parse(localStorage.getItem("user"));
+  const [user, setUser] = useState(storedUser ? storedUser : null);
```

Comment:
There is no need to create a state variable named user, in the scope of this component you only need the user object to use its ID when doing a request on the API endpoint.

Reply:
done